site/profile/manifests/efa.pp

```diff
@@ -0,0 +1,41 @@
+class profile::efa {
+  package { 'libibverbs-utils':
+    ensure => 'installed',
+  }
+
+  package { 'rdma-core-devel':
+    ensure => 'installed',
+  }
+
+  package { 'librdmacm-utils':
+    ensure => 'installed',
+  }
+
+  exec { 'download-efa-driver':
```

Comment:
Unless there is something I am missing, this resource will execute at each Puppet apply since to avoid running, it would look for `/tmp/aws-efa-installer` which is deleted by the resource `Tidy['delete-efa-driver']`.

To avoid this, use an archive resource instead. This will keep the tarball in a local cache folder, but that's a small price to pay. You can find example here : https://github.com/ComputeCanada/puppet-magic_castle/blob/main/site/profile/manifests/accounts.pp#L81

Reply:
True! I added the `tidy` part after and I forgot that it would remove the condition on the exec. I will use the "archive" resource.